### ISSUE TITLE
Creating install targets

### ISSUE THREAD

**Issue by kevans91** [sirjuddington/SLADE#371]
Hello!

As I've mentioned elsewhere, I'd like to create a FreeBSD port for SLADE. With the merging of [this pull request](https://github.com/sirjuddington/SLADE/pull/370), this is now feasible. I've since convinced the maintainer of `x11-toolkits` to include an option for enabling the wx stackwalker, and this is in the process of happening. Stackwalker disabled will still be the default, at the very least until FreeBSD 9-STABLE and 9.3-RELEASE support is dropped at the end of '16. I don't suspect this will be too much of an issue, though.

As for the subject of this issue, though. I would like to (or get someone else to) create install targets for SLADE, at the very least on *NIX-based systems. From what I could see, this wouldn't require too many changes. I believe the ideal solution would install the binary at `$PREFIX/bin/slade` and slade.pk3 at `$PREFIX/share/slade/slade.pk3`. `~/.slade3/` for configuration files/logs is already perfect, so it would seem that the main change here would be also searching `$PREFIX/share/slade/slade.pk3` for the pk3 file. This could be not too rough with wxStandardPaths::(Get/Set)InstallPrefix, I suspect.

Thoughts/comments/concerns?

**Comment by Gaerzi** [comment]
GetInstallPrefix() only exists on Unix versions of wxWidgets. Would GetResourcesDir() work? That's honestly the most appropriate place to put resources anyway.

**Comment by kevans91** [comment]
Ah, true, GetResourcesDir() is quite appropriate. I somehow missed that one when I was cruising the documentation. =P

**Comment by kevans91** [comment]
I went ahead and started some work on this on my `Installable` [branch](https://github.com/kevans91/SLADE/tree/installable) if anyone would like to review this while it's in progress.

I'm not entirely sure what the most appropriate way to handle APPLE installations would be, so I've only set this up for not APPLE. I'm assuming it won't be used on WIN32, so the INSTALL() stuff shouldn't hurt much there. If it's important to omit it, though, then that's fair enough.

I've used SetInstallPrefix() only #if defined(__UNIX__) and if INSTALL_PREFIX is set, because it does affect wxStandardPaths::GetResourcesDir() in a desirable way for UNIX machines. I used INSTALL_PREFIX as the macro to be defined within SLADE if CMAKE_INSTALL_PREFIX is set so that it's not dependent on a particular build/configuration system.

So far, this all builds and installs correctly. I have the FreeBSD port for this now built and that works all hunky-dory. All I think I have left to do is verify that I setup the "slade.pk3" finding addition right (I haven't actually ran the binary that it produced yet, as I've not an X server handy in my current location), and that it didn't affect any of the current use-cases (which it shouldn't have, as far as I can tell -- it still does all of the slade.pk3 fall-back correctly, and install() stuff won't get ran unless 'make install' is invoked).

**Comment by Gaerzi** [comment]
That's basically how I would have done it. The changes in Main.h/MainApp.cpp are fine as far as I'm concerned.

**Comment by kevans91** [comment]
Alright, cool. I did test it when I got home, both with my system's standard installation prefix (/usr/local) as well as a non-standard prefix (/usr/iso-test -- not sure why I chose this name). Both worked fine and off of the slade.pk3 in the resource directory (based off of the prefix). Leaving the binary and slade.pk3 in the same directory also worked just fine, so current usage should not be affected even in the slightest. This is also nice, because presumably other *nix-based (with obvious exception to OS, from what it seems) OS's and their packaging systems will also find this useful, if one so chooses to take it to them.

I may let this sit and sizzle for a day or two (unless you recommend otherwise) to see if anyone has any objections to the idea (although, I don't know why I seem to think anyone would).
